gcivil-nyu-org/Wednesday-Fall2023-Team-1
# Update README.md and Address Spotify Token Cache Issue

### Title: Update README.md and Address Spotify Token Issue

### Message:

#### Overview
This pull request introduces several enhancements and a bug fix related to our usage of the Spotify and Genius APIs. These changes include updates to the documentation in `README.md` and a fix related to the handling of Spotify tokens in our application.

#### 📜 Documentation Updates
- **README.md Enhancements**:
  - Added detailed instructions for configuring the Python environment and installing dependencies.
  - Introduced guidelines for managing and using environment variables, specifically for API credentials.
  - Expanded upon the interaction with the Genius API, illustrating usage with code snippets.
  - Updated the list of required packages and installation instructions.
  - Provided additional guidance on utilizing the `.env` file for local development and production configurations.

- **.gitignore Updates**:
  - Updated to exclude specific files and directories related to Visual Studio Code, enhancing developer experience without affecting the repository.

- Commits:
  - [8194cef](https://github.com/BoyanXu/Wednesday-Fall2023-Team-1/commit/8194cef7f5aeaf6e61a910dcd0a677b0bd73700e)
  - [f15c318](https://github.com/BoyanXu/Wednesday-Fall2023-Team-1/commit/f15c3188985ffb897b6a8026251907a799b439bd)

#### 🐛 Bug Fix: Spotify Token Caching Issue
- **Issue**: The cached Spotify token was leading to an `invalid_client` error.
- **Resolution**: Modified the `get_spotify_token` function to set `check_cache=False`, circumventing the `SpotifyOauthError`.

- Commit:
  - [0a61a1d](https://github.com/BoyanXu/Wednesday-Fall2023-Team-1/commit/0a61a1d5bc2fe0e64cc34cc3caa337d1169a2a66)

#### 🔧 Chore: Manage API Credentials Securely
- **Enhancements**:
  - Transitioned to loading API credentials (`CLIENT_ID` and `CLIENT_SECRET`) from environment variables, augmenting security by avoiding hardcoding.
  - Added a `.env.example` file to guide developers in setting their environment variables.
  - Ensured that API credentials are not tracked in version control.

- Commit:
  - [f0875a7](https://github.com/BoyanXu/Wednesday-Fall2023-Team-1/commit/f0875a7ec0553944519fd94d83c703648becbcef)

#### Testing
Please ensure to test the following:
- Confirm that the application correctly interfaces with the Spotify and Genius APIs using credentials stored in environment variables.
- Validate that the README.md provides clear and accurate setup and configuration instructions.
- Ensure that the fix resolves the `invalid_client` error when interacting with the Spotify API.

#### Notes to Reviewers
- Please verify the correctness and clarity of the instructions provided in `README.md`.
- Validate the bug fix in various scenarios to ensure robustness.
- Feedback on the clarity and comprehensiveness of the documentation is highly appreciated.

#### Checklist
- [x] I have ensured that my code follows the style guidelines of this project.
